Mailbox/Program.cs

```diff
@@ -87,19 +87,72 @@ static void Main(string[] args)
             }
         }
 
-        public static string GetOwnersDisplay(Mailboxes mailboxes)
+        public static string? GetOwnersDisplay(Mailboxes mailboxes)
         {
-            
+            HashSet<Person> owners = new HashSet<Person>();
+
+            foreach(Mailbox mailbox in mailboxes)
+            {
+                owners.Add(mailbox.Owner);
+            }
+
+            string output = "";
+            foreach(Person owner in owners)
+            {
+                output += owner.ToString() + '\n';
+            }
+
+            return output;
         }
 
-        public static string GetMailboxDetails(Mailboxes mailboxes, int x, int y)
+        public static string? GetMailboxDetails(Mailboxes mailboxes, int x, int y)
         {
-            
+            foreach(Mailbox mailbox in mailboxes)
+            {
+                if(mailbox.Location == (x, y))
+                {
+                    return mailbox.ToString();
+                }
+            }
+
+            return null;
         }
 
-        public static Mailbox AddNewMailbox(Mailboxes mailboxes, string firstName, string lastName, Size size)
+        public static Mailbox? AddNewMailbox(Mailboxes mailboxes, string firstName, string lastName, Size size)
         {
-            
+            int width = mailboxes.Width;
+            int height = mailboxes.Height;
+
+            for (int i = 0; i < width; i++)
+            {
+                for (int a = 0; a < height; i++)
+                {
+                    if (!mailboxes.GetAdjacentPeople(i, a, out HashSet<Person> personSet))
+                    {
+                        Boolean valid = true;
```

Comment:
Use keyword over type name.
```suggestion
                        bool valid = true;
```